woodpecker/log/log_handle.go

```diff
@@ -169,6 +171,31 @@ func (l *logHandleImpl) GetSegments(ctx context.Context) (map[int64]*meta.Segmen
 	return result, err
 }
 
+// OpenLogWriterAdaptively opens an appropriate log writer based on fence policy configuration.
+// Uses distributed lock writer when condition write is disabled, otherwise uses internal writer with condition write.
+func (l *logHandleImpl) OpenLogWriterAdaptively(ctx context.Context) (LogWriter, error) {
+	fencePolicy := l.cfg.Woodpecker.Logstore.FencePolicy
+
+	// Use distributed lock writer when condition write is disabled
+	if l.cfg.Woodpecker.Storage.IsStorageMinio() && fencePolicy.IsConditionWriteDisabled() {
+		logger.Ctx(ctx).Info("opening log writer with distributed locks (condition write disabled)",
+			zap.String("logName", l.Name),
+			zap.Int64("logId", l.Id))
+		return l.OpenLogWriter(ctx)
+	}
+
+	// Use internal writer with condition write support
+	logger.Ctx(ctx).Info("opening internal log writer with condition write support",
+		zap.String("logName", l.Name),
+		zap.Int64("logId", l.Id),
+		zap.String("fencePolicy", fencePolicy.ConditionWrite))
+	return l.OpenInternalLogWriter(ctx)
+}
+
+// OpenLogWriter opens a log writer with distributed locks for the log.
```

Comment:
openLogWriter should support one config, fencePolicy.IsConditionWriteDisabled(), no need to create two functions.

Reply:
ok